packages/reactive-element/src/decorators/base.ts

```diff
@@ -47,8 +47,21 @@ export const standardPrototypeMethod = (
   descriptor,
 });
 
+/**
+ * Helper for decorating a property that is compatible with both TypeScript
+ * and Babel decorators. The optional `finisher` can be used to perform work on
+ * the class. It . The optional `descriptor` should return a PropertyDescriptor
+ * to install for the given property.
+ *
+ * @param finisher {(ctor: typeof ReactiveElement, property: PropertyKey) => void)} Optional finisher
+ * @param descriptor {(property: PropertyKey) => PropertyDescriptor} Optional descriptor generator
+ * @returns {ClassElement|void}
+ */
 export const decorateProperty = (
-  decorator: (ctor: typeof ReactiveElement, property: PropertyKey) => void
+  finisher?:
+    | ((ctor: typeof ReactiveElement, property: PropertyKey) => void)
+    | null,
+  descriptor?: (property: PropertyKey) => PropertyDescriptor
```

Comment:
Since these options are basically mutually exclusive in practice, API-wise seems better as a single a keyed bag `{finisher: ..., descriptor: ...}`.

Reply:
Done.